x/tx/signing/context_test.go

```diff
@@ -145,6 +145,31 @@ func TestGetSigners(t *testing.T) {
 	}
 }
 
+func TestDefineCustomGetSigners(t *testing.T) {
+	customMsg := &testpb.Ballot{}
+	signers := [][]byte{[]byte("foo")}
+	options := Options{
+		AddressCodec:          dummyAddressCodec{},
+		ValidatorAddressCodec: dummyValidatorAddressCodec{},
+	}
+	context, err := NewContext(options)
+	require.NoError(t, err)
+
+	_, err = context.GetSigners(customMsg)
+	// without a custom signer we should get an error
+	require.ErrorContains(t, err, "use DefineCustomGetSigners to specify")
+
+	// create a new context with a custom signer
```

Comment:
Just curious, can you overwrite the proto annotation by defining a custom get signer?
Can we have a test with the expected result?

Reply:
That's a good point.  I think the annotation would override the custom signer but I'll definitely make a test for it.